### ISSUE TITLE
Hail on windows

### ISSUE THREAD

**Issue by iris-garden** [iris-garden/test-process#2835]
<h2>Zaki said:</h2>
<p>I am currently using windows and wanted to use hail to run gwas and i downloaded python java and spark and set their paths as well and after importing hail, i wanted to initialize hail and got this error i have tried everything but failed. can anyone help please.</p>
<p>import hail as hl<br>
hl.init()</p>
<p>error</p>
<p>Py4JJavaError: An error occurred while calling z:is.hail.backend.spark.SparkBackend.apply.<br>
: java.lang.NoSuchMethodError: scala.Predef$.refArrayOps([Ljava/lang/Object;)Lscala/collection/mutable/ArrayOps;<br>
at is.hail.backend.spark.SparkBackend$.majorMinor$1(SparkBackend.scala:65)<br>
at is.hail.backend.spark.SparkBackend$.checkSparkCompatibility(SparkBackend.scala:67)<br>
at is.hail.backend.spark.SparkBackend$.createSparkConf(SparkBackend.scala:78)<br>
at is.hail.backend.spark.SparkBackend$.configureAndCreateSparkContext(SparkBackend.scala:127)<br>
at is.hail.backend.spark.SparkBackend$.apply(SparkBackend.scala:203)<br>
at is.hail.backend.spark.SparkBackend.apply(SparkBackend.scala)<br>
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br>
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)<br>
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)<br>
at java.lang.reflect.Method.invoke(Unknown Source)<br>
at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)<br>
at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)<br>
at py4j.Gateway.invoke(Gateway.java:282)<br>
at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)<br>
at py4j.commands.CallCommand.execute(CallCommand.java:79)<br>
at py4j.GatewayConnection.run(GatewayConnection.java:238)<br>
at java.lang.Thread.run(Unknown Source</p>

<h2>danking said:</h2>
<p>Hi Zaki ,</p>
<p>I’m sorry you’re having trouble. We don’t officially support Windows. I suspect many things will not work. I can try to help you get <code>hl.init</code> to work.</p>
<p>Are you using the Windows Subsystem for Linux? What is the output of this:</p>

```python
python --version
java -version
echo %SPARK_HOME%
echo %PYTHONPATH%
which python
which java
python -m pip show hail
python -m pip show pyspark

```

<h2>Zaki said:</h2>
<p><strong>echo %PYTHONPATH%</strong><br>
%PYTHONPATH%</p>
<p>it was just returnig pytpath so I tried</p>
<p><strong>where python</strong><br>
C:\Users\Zaki\anaconda3\python.exe</p>
<p><strong>which python</strong><br>
/c/Users/Zaki/anaconda3/python</p>

<h2>Zaki said:</h2>
<p>Thank you Dan, for the response.</p>
<p><strong>python --version</strong><br>
Python 3.7.10</p>
<p><strong>java -version</strong><br>
java version “1.8.0_291”<br>
Java™ SE Runtime Environment (build 1.8.0_291-b09)<br>
Java HotSpot™ 64-Bit Server VM (build 25.291-b09, mixed mode)</p>
<p><strong>echo %SPARK_HOME%</strong><br>
C:\Spark\spark-3.1.1-bin-hadoop2.7</p>
<p><strong>echo %PYTHONPATH%</strong><br>
%PYTHONPATH%</p>
<p>it was just returnig pytpath so I tried</p>
<p><strong>where python</strong><br>
C:\Users\Zaki\anaconda3\python.exe</p>
<p><strong>which python</strong><br>
/c/Users/Zaki/anaconda3/python</p>
<p><strong>which java</strong><br>
/c/Program Files (x86)/Common Files/Oracle/Java/javapath/java</p>
<p><strong>python -m pip show hail</strong><br>
Name: hail<br>
Version: 0.2.64<br>
Summary: Scalable library for exploring and analyzing genomic data.<br>
Home-page: <a href="https://hail.is/" rel="noopener nofollow ugc">https://hail.is</a><br>
Author: Hail Team<br>
Author-email: <a href="mailto:hail@broadinstitute.org">hail@broadinstitute.org</a><br>
License: UNKNOWN<br>
Location: c:\users\zaki\anaconda3\lib\site-packages<br>
Requires: parsimonious, aiohttp, decorator, asyncinit, requests, hurry.filesize, pandas, tqdm, aiohttp-session, google-cloud-storage, gcsfs, PyJWT, Deprecated, humanize, numpy, bokeh, dill, pyspark, python-json-logger, nest-asyncio, tabulate, scipy<br>
Required-by:</p>
<p><strong>python -m pip show pyspark</strong><br>
Name: pyspark<br>
Version: 2.4.1<br>
Summary: Apache Spark Python API<br>
Home-page: <a href="https://github.com/apache/spark/tree/master/python" rel="noopener nofollow ugc">spark/python at master · apache/spark · GitHub</a><br>
Author: Spark Developers<br>
Author-email: <a href="mailto:dev@spark.apache.org">dev@spark.apache.org</a><br>
License: <a href="http://www.apache.org/licenses/LICENSE-2.0" rel="noopener nofollow ugc">http://www.apache.org/licenses/LICENSE-2.0</a><br>
Location: c:\users\zaki\anaconda3\lib\site-packages<br>
Requires: py4j<br>
Required-by: hail</p>

<h2>danking said:</h2>

<blockquote>
<p><strong>echo %SPARK_HOME%</strong></p>
</blockquote>

<p>This is the problem. Your spark home points at spark 3.1.1 but your pyspark is 2.4.1. How did you install Hail? Can you unset SPARK_HOME?</p>

<h2>Zaki said:</h2>
<p>Thank you very much Dan, Its working now.</p>

<h2>Zaki said:</h2>
<p>Sorry to disturb you again but now I am facing another issue</p>
<p><strong>hl.plot.output_notebook()</strong><br>
BokehJS 1.4.0 successfully loaded.</p>
<p><strong>hl.utils.get_1kg(‘data/’)</strong><br>
Initializing Hail with default parameters… Running on Apache Spark version 2.4.1 SparkUI available at <a href="http://desktop-llovftk:4040/" rel="noopener nofollow ugc">http://DESKTOP-LLOVFTK:4040</a> Welcome to __ __ <>__ / /<em>/ /__ __/ / / __ / _ `/ / / /</em>/ /<em>/_,</em>/<em>/</em>/ version 0.2.64-1ef70187dc78 LOGGING: writing to C:\Users\Zaki\hail-20210331-1328-0.2.64-1ef70187dc78.log 2021-03-31 13:28:45 Hail: INFO: downloading 1KG VCF … Source: <a href="https://storage.googleapis.com/hail-tutorial/1kg.vcf.bgz" rel="noopener nofollow ugc">https://storage.googleapis.com/hail-tutorial/1kg.vcf.bgz</a></p>
<p>FatalError: IllegalArgumentException: Wrong FS: file://C:\Users\Zaki\AppData\Local\Temp\tmpq_djf_jt\1kg.vcf.bgz, expected: file:///</p>
<p>Java stack trace:<br>
java.lang.IllegalArgumentException: Wrong FS: file://C:\Users\Zaki\AppData\Local\Temp\tmpq_djf_jt\1kg.vcf.bgz, expected: file:///<br>
at org.apache.hadoop.fs.FileSystem.checkPath(FileSystem.java:649)<br>
at org.apache.hadoop.fs.RawLocalFileSystem.pathToFile(RawLocalFileSystem.java:82)<br>
at org.apache.hadoop.fs.RawLocalFileSystem.deprecatedGetFileStatus(RawLocalFileSystem.java:606)<br>
at org.apache.hadoop.fs.RawLocalFileSystem.getFileLinkStatusInternal(RawLocalFileSystem.java:824)<br>
at org.apache.hadoop.fs.RawLocalFileSystem.getFileStatus(RawLocalFileSystem.java:601)<br>
at org.apache.hadoop.fs.FilterFileSystem.getFileStatus(FilterFileSystem.java:421)<br>
at org.apache.hadoop.fs.ChecksumFileSystem$ChecksumFSInputChecker.(ChecksumFileSystem.java:142)<br>
at org.apache.hadoop.fs.ChecksumFileSystem.open(ChecksumFileSystem.java:346)<br>
at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:769)<br>
at is.hail.io.fs.HadoopFS.openNoCompression(HadoopFS.scala:83)<br>
at is.hail.io.fs.FS$class.copy(FS.scala:188)<br>
at is.hail.io.fs.HadoopFS.copy(HadoopFS.scala:70)<br>
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br>
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)<br>
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)<br>
at java.lang.reflect.Method.invoke(Unknown Source)<br>
at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)<br>
at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)<br>
at py4j.Gateway.invoke(Gateway.java:282)<br>
at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)<br>
at py4j.commands.CallCommand.execute(CallCommand.java:79)<br>
at py4j.GatewayConnection.run(GatewayConnection.java:238)<br>
at java.lang.Thread.run(Unknown Source)</p>
<p>Hail version: 0.2.64-1ef70187dc78<br>
Error summary: IllegalArgumentException: Wrong FS: file://C:\Users\Zaki\AppData\Local\Temp\tmpq_djf_jt\1kg.vcf.bgz, expected: file:///</p>

<h2>danking said:</h2>
<p>I think the only way to make this work is to install and use Hail inside the <a href="https://docs.microsoft.com/en-us/windows/wsl/about">Windows Subsystem for Linux</a></p>

<h2>Zaki said:</h2>
<p>So, I installed ubuntu on my Virtual Machine and installed hail and other applications as java, g++ etc<br>
and also set the paths for spark and java<br>
when i run the command in jupyter notebook</p>
<p><strong>import hail as hl</strong><br>
i get the following error</p>
<p><strong>---------------------------------------------------------------------------</strong><br>
<strong>TypeError                                 Traceback (most recent call last)</strong><br>
<strong> in </strong><br>
<strong>----> 1 import hail as hl</strong></p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/hail/<strong>init</strong>.py in </strong><br>
**     32 # F401 ‘.expr.*’ imported but unused**<br>
**     33 # E402 module level import not at top of file**<br>
<strong>—> 34 from .table import Table, GroupedTable, asc, desc  # noqa: E402</strong><br>
**     35 from .matrixtable import MatrixTable, GroupedMatrixTable  # noqa: E402**<br>
**     36 from .expr import *  # noqa: F401,F403,E402**</p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/hail/table.py in </strong><br>
**      2 import itertools**<br>
**      3 import pandas**<br>
<strong>----> 4 import pyspark</strong><br>
**      5 from typing import Optional, Dict, Callable**<br>
**      6 **</p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/pyspark/<strong>init</strong>.py in </strong><br>
**     49 **<br>
**     50 from pyspark.conf import SparkConf**<br>
<strong>—> 51 from pyspark.context import SparkContext</strong><br>
**     52 from pyspark.rdd import RDD, RDDBarrier**<br>
**     53 from pyspark.files import SparkFiles**</p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/pyspark/context.py in </strong><br>
**     29 from py4j.protocol import Py4JError**<br>
**     30 **<br>
<strong>—> 31 from pyspark import accumulators</strong><br>
**     32 from pyspark.accumulators import Accumulator**<br>
**     33 from pyspark.broadcast import Broadcast, BroadcastPickleRegistry**</p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/pyspark/accumulators.py in </strong><br>
**     95     import socketserver as SocketServer**<br>
**     96 import threading**<br>
<strong>—> 97 from pyspark.serializers import read_int, PickleSerializer</strong><br>
**     98 **<br>
**     99 **</p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/pyspark/serializers.py in </strong><br>
**     69     xrange = range**<br>
**     70 **<br>
<strong>—> 71 from pyspark import cloudpickle</strong><br>
**     72 from pyspark.util import _exception_message**<br>
**     73 **</p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/pyspark/cloudpickle.py in </strong><br>
**    143 **<br>
**    144 **<br>
<strong> → 145 _cell_set_template_code = _make_cell_set_template_code()</strong><br>
**    146 **<br>
**    147 **</p>
<p><strong>~/anaconda3/lib/python3.8/site-packages/pyspark/cloudpickle.py in _make_cell_set_template_code()</strong><br>
**    124         )**<br>
**    125     else:**<br>
<strong> → 126         return types.CodeType(</strong><br>
**    127             co.co_argcount,**<br>
**    128             co.co_kwonlyargcount,**</p>
<p><strong>TypeError: an integer is required (got type bytes)</strong></p>

<h2>danking said:</h2>
<p>Hey Zaki , unfortunately, the currently released version of Hail does not support Python 3.8 because PySpark (a library on which we depend) does not support Python 3.8. Try <a href="https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-with-commands">creating a Python 3.7 environment using Anaconda</a>.</p>

<h2>Zaki said:</h2>
<p>Thank you danking its working now but now when I run this command it shows me this error</p>
<p>hl.utils.get_1kg(’/data’)</p>
<p><strong>021-04-05 21:34:58 Hail: INFO: downloading 1KG VCF …</strong><br>
**  Source: <a href="https://storage.googleapis.com/hail-tutorial/1kg.vcf.bgz**" rel="noopener nofollow ugc">https://storage.googleapis.com/hail-tutorial/1kg.vcf.bgz**</a><br>
<strong>2021-04-05 21:35:00 Hail: INFO: importing VCF and writing to matrix table…</strong><br>
<strong>2021-04-05 21:35:07 Hail: INFO: Coerced sorted dataset</strong></p>
<p><strong>---------------------------------------------------------------------------</strong><br>
<strong>FatalError                                Traceback (most recent call last)</strong><br>
<strong> in </strong><br>
<strong>----> 1 hl.utils.get_1kg(’/data’)</strong></p>
<p><strong>~/anaconda3/lib/python3.7/site-packages/hail/utils/tutorial.py in get_1kg(output_dir, overwrite)</strong><br>
**     80         cluster_readable_vcf = _copy_to_tmp(fs, local_path_uri(tmp_vcf), extension=‘vcf.bgz’)**<br>
**     81         info(‘importing VCF and writing to matrix table…’)**<br>
<strong>—> 82         hl.import_vcf(cluster_readable_vcf, min_partitions=16).write(matrix_table_path, overwrite=True)</strong><br>
**     83 **<br>
**     84         tmp_sample_annot = os.path.join(tmp_dir, ‘1kg_annotations.txt’)**</p>
<p><strong> in write(self, output, overwrite, stage_locally, _codec_spec, _partitions)</strong></p>
<p><em><em>~/anaconda3/lib/python3.7/site-packages/hail/typecheck/check.py in wrapper(__original_func, <em>args, <strong>kwargs)</strong><br>
**    575     def wrapper(<em><em>original_func, *args, <strong>kwargs):</strong><br>
**    576         args</em>, kwargs</em> = check_all(__original_func, args, kwargs, checkers, is_method=is_method)</em></em><br>
<strong> → 577         return <em><em>original_func(*args</em>, **kwargs</em>)</strong><br>
**    578 **<br>
**    579     return wrapper</em>*</p>
<p><strong>~/anaconda3/lib/python3.7/site-packages/hail/matrixtable.py in write(self, output, overwrite, stage_locally, _codec_spec, _partitions)</strong><br>
**   2526 **<br>
**   2527         writer = ir.MatrixNativeWriter(output, overwrite, stage_locally, _codec_spec, _partitions, _partitions_type)**<br>
<strong> → 2528         Env.backend().execute(ir.MatrixWrite(self._mir, writer))</strong><br>
**   2529 **<br>
**   2530     class _Show:**</p>
<p><strong>~/anaconda3/lib/python3.7/site-packages/hail/backend/py4j_backend.py in execute(self, ir, timed)</strong><br>
**     96                 raise HailUserError(message_and_trace) from None**<br>
**     97 **<br>
<strong>—> 98             raise e</strong></p>
<p><strong>~/anaconda3/lib/python3.7/site-packages/hail/backend/py4j_backend.py in execute(self, ir, timed)</strong><br>
**     72         # print(self._hail_package.expr.ir.Pretty.apply(jir, True, -1))**<br>
**     73         try:**<br>
<strong>—> 74             result = json.loads(self._jhc.backend().executeJSON(jir))</strong><br>
**     75             value = ir.typ._from_json(result[‘value’])**<br>
**     76             timings = result[‘timings’]**</p>
<p><em><em>~/anaconda3/lib/python3.7/site-packages/py4j/java_gateway.py in <strong>call</strong>(self, <em>args)</em></em><br>
**   1255         answer = self.gateway_client.send_command(command)</em>*<br>
**   1256         return_value = get_return_value(**<br>
<strong> → 1257             answer, self.gateway_client, self.target_id, self.name)</strong><br>
**   1258 **<br>
**   1259         for temp_arg in temp_args:**</p>
<p><em><em>~/anaconda3/lib/python3.7/site-packages/hail/backend/py4j_backend.py in deco(<em>args, <strong>kwargs)</strong><br>
**     30                 raise FatalError(’%s\n\nJava stack trace:\n%s\n’</em></em><br>
**     31                                  ‘Hail version: %s\n’</em>*<br>
<strong>—> 32                                  ‘Error summary: %s’ % (deepest, full, hail.<strong>version</strong>, deepest), error_id) from None</strong><br>
**     33         except pyspark.sql.utils.CapturedException as e:**<br>
**     34             raise FatalError(’%s\n\nJava stack trace:\n%s\n’**</p>
<p><strong>FatalError: IOException: Mkdirs failed to create /data/1kg.mt/rows/rows/parts (exists=false, cwd=file:/home/zaki)</strong></p>
<p><strong>Java stack trace:</strong><br>
<strong>org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1.0 (TID 16, localhost, executor driver): java.io.IOException: Mkdirs failed to create /data/1kg.mt/rows/rows/parts (exists=false, cwd=file:/home/zaki)</strong><br>
**	at org.apache.hadoop.fs.ChecksumFileSystem.create(ChecksumFileSystem.java:455)**<br>
**	at org.apache.hadoop.fs.ChecksumFileSystem.create(ChecksumFileSystem.java:440)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:911)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:892)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:789)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:778)**<br>
**	at is.hail.io.fs.HadoopFS.createNoCompression(HadoopFS.scala:74)**<br>
**	at is.hail.io.fs.FS$class.create(FS.scala:151)**<br>
**	at is.hail.io.fs.HadoopFS.create(HadoopFS.scala:70)**<br>
**	at is.hail.io.RichContextRDDRegionValue$.writeSplitRegion(RichContextRDDRegionValue.scala:106)**<br>
**	at is.hail.rvd.RVD$$anonfun$29.apply(RVD.scala:938)**<br>
**	at is.hail.rvd.RVD$$anonfun$29.apply(RVD.scala:936)**<br>
**	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitionsWithIndex$1$$anonfun$apply$18.apply(ContextRDD.scala:259)**<br>
**	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitionsWithIndex$1$$anonfun$apply$18.apply(ContextRDD.scala:259)**<br>
**	at is.hail.utils.richUtils.RichContextRDD$$anonfun$cleanupRegions$1$$anonfun$2.apply(RichContextRDD.scala:62)**<br>
**	at is.hail.utils.richUtils.RichContextRDD$$anonfun$cleanupRegions$1$$anonfun$2.apply(RichContextRDD.scala:62)**<br>
**	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:435)**<br>
**	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:441)**<br>
**	at is.hail.utils.richUtils.RichContextRDD$$anonfun$cleanupRegions$1$$anon$1.hasNext(RichContextRDD.scala:71)**<br>
**	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:439)**<br>
**	at scala.collection.Iterator$class.foreach(Iterator.scala:891)**<br>
**	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)**<br>
**	at scala.collection.generic.Growable$class.$plus$plus$eq(Growable.scala:59)**<br>
**	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:104)**<br>
**	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:48)**<br>
**	at scala.collection.TraversableOnce$class.to(TraversableOnce.scala:310)**<br>
**	at scala.collection.AbstractIterator.to(Iterator.scala:1334)**<br>
**	at scala.collection.TraversableOnce$class.toBuffer(TraversableOnce.scala:302)**<br>
**	at scala.collection.AbstractIterator.toBuffer(Iterator.scala:1334)**<br>
**	at scala.collection.TraversableOnce$class.toArray(TraversableOnce.scala:289)**<br>
**	at scala.collection.AbstractIterator.toArray(Iterator.scala:1334)**<br>
**	at org.apache.spark.rdd.RDD$$anonfun$collect$1$$anonfun$13.apply(RDD.scala:945)**<br>
**	at org.apache.spark.rdd.RDD$$anonfun$collect$1$$anonfun$13.apply(RDD.scala:945)**<br>
**	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)**<br>
**	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)**<br>
**	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)**<br>
**	at org.apache.spark.scheduler.Task.run(Task.scala:121)**<br>
**	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:403)**<br>
**	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)**<br>
**	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:409)**<br>
**	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)**<br>
**	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)**<br>
**	at java.lang.Thread.run(Thread.java:748)**</p>
<p><strong>Driver stacktrace:</strong><br>
**	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1889)**<br>
**	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1877)**<br>
**	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1876)**<br>
**	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)**<br>
**	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)**<br>
**	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1876)**<br>
**	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)**<br>
**	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:926)**<br>
**	at scala.Option.foreach(Option.scala:257)**<br>
**	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:926)**<br>
**	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2110)**<br>
**	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2059)**<br>
**	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2048)**<br>
**	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)**<br>
**	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:737)**<br>
**	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2061)**<br>
**	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2082)**<br>
**	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2101)**<br>
**	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)**<br>
**	at org.apache.spark.rdd.RDD$$anonfun$collect$1.apply(RDD.scala:945)**<br>
**	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)**<br>
**	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)**<br>
**	at org.apache.spark.rdd.RDD.withScope(RDD.scala:363)**<br>
**	at org.apache.spark.rdd.RDD.collect(RDD.scala:944)**<br>
**	at is.hail.sparkextras.ContextRDD.collect(ContextRDD.scala:176)**<br>
**	at is.hail.rvd.RVD.writeRowsSplit(RVD.scala:953)**<br>
**	at is.hail.expr.ir.MatrixValue.write(MatrixValue.scala:246)**<br>
**	at is.hail.expr.ir.MatrixNativeWriter.apply(MatrixWriter.scala:62)**<br>
**	at is.hail.expr.ir.WrappedMatrixWriter.apply(MatrixWriter.scala:41)**<br>
**	at is.hail.expr.ir.Interpret$.run(Interpret.scala:819)**<br>
**	at is.hail.expr.ir.Interpret$.alreadyLowered(Interpret.scala:53)**<br>
**	at is.hail.expr.ir.InterpretNonCompilable$.interpretAndCoerce$1(InterpretNonCompilable.scala:16)**<br>
**	at is.hail.expr.ir.InterpretNonCompilable$.is$hail$expr$ir$InterpretNonCompilable$$rewrite$1(InterpretNonCompilable.scala:53)**<br>
**	at is.hail.expr.ir.InterpretNonCompilable$.apply(InterpretNonCompilable.scala:58)**<br>
**	at is.hail.expr.ir.lowering.InterpretNonCompilablePass$.transform(LoweringPass.scala:67)**<br>
**	at is.hail.expr.ir.lowering.LoweringPass$$anonfun$apply$3$$anonfun$1.apply(LoweringPass.scala:15)**<br>
**	at is.hail.expr.ir.lowering.LoweringPass$$anonfun$apply$3$$anonfun$1.apply(LoweringPass.scala:15)**<br>
**	at is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:81)**<br>
**	at is.hail.expr.ir.lowering.LoweringPass$$anonfun$apply$3.apply(LoweringPass.scala:15)**<br>
**	at is.hail.expr.ir.lowering.LoweringPass$$anonfun$apply$3.apply(LoweringPass.scala:13)**<br>
**	at is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:81)**<br>
**	at is.hail.expr.ir.lowering.LoweringPass$class.apply(LoweringPass.scala:13)**<br>
**	at is.hail.expr.ir.lowering.InterpretNonCompilablePass$.apply(LoweringPass.scala:62)**<br>
**	at is.hail.expr.ir.lowering.LoweringPipeline$$anonfun$apply$1.apply(LoweringPipeline.scala:14)**<br>
**	at is.hail.expr.ir.lowering.LoweringPipeline$$anonfun$apply$1.apply(LoweringPipeline.scala:12)**<br>
**	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)**<br>
**	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)**<br>
**	at is.hail.expr.ir.lowering.LoweringPipeline.apply(LoweringPipeline.scala:12)**<br>
**	at is.hail.expr.ir.CompileAndEvaluate$._apply(CompileAndEvaluate.scala:28)**<br>
**	at is.hail.backend.spark.SparkBackend.is$hail$backend$spark$SparkBackend$$_execute(SparkBackend.scala:362)**<br>
**	at is.hail.backend.spark.SparkBackend$$anonfun$execute$1.apply(SparkBackend.scala:346)**<br>
**	at is.hail.backend.spark.SparkBackend$$anonfun$execute$1.apply(SparkBackend.scala:343)**<br>
**	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1$$anonfun$apply$1.apply(ExecuteContext.scala:48)**<br>
**	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1$$anonfun$apply$1.apply(ExecuteContext.scala:48)**<br>
**	at is.hail.utils.package$.using(package.scala:618)**<br>
**	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:48)**<br>
**	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:47)**<br>
**	at is.hail.utils.package$.using(package.scala:618)**<br>
**	at is.hail.annotations.RegionPool$.scoped(RegionPool.scala:13)**<br>
**	at is.hail.expr.ir.ExecuteContext$.scoped(ExecuteContext.scala:47)**<br>
**	at is.hail.backend.spark.SparkBackend.withExecuteContext(SparkBackend.scala:256)**<br>
**	at is.hail.backend.spark.SparkBackend.execute(SparkBackend.scala:343)**<br>
**	at is.hail.backend.spark.SparkBackend$$anonfun$7.apply(SparkBackend.scala:387)**<br>
**	at is.hail.backend.spark.SparkBackend$$anonfun$7.apply(SparkBackend.scala:385)**<br>
**	at is.hail.utils.ExecutionTimer$.time(ExecutionTimer.scala:52)**<br>
**	at is.hail.backend.spark.SparkBackend.executeJSON(SparkBackend.scala:385)**<br>
**	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)**<br>
**	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)**<br>
**	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)**<br>
**	at java.lang.reflect.Method.invoke(Method.java:498)**<br>
**	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)**<br>
**	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)**<br>
**	at py4j.Gateway.invoke(Gateway.java:282)**<br>
**	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)**<br>
**	at py4j.commands.CallCommand.execute(CallCommand.java:79)**<br>
**	at py4j.GatewayConnection.run(GatewayConnection.java:238)**<br>
**	at java.lang.Thread.run(Thread.java:748)**</p>
<p><strong>java.io.IOException: Mkdirs failed to create /data/1kg.mt/rows/rows/parts (exists=false, cwd=file:/home/zaki)</strong><br>
**	at org.apache.hadoop.fs.ChecksumFileSystem.create(ChecksumFileSystem.java:455)**<br>
**	at org.apache.hadoop.fs.ChecksumFileSystem.create(ChecksumFileSystem.java:440)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:911)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:892)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:789)**<br>
**	at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:778)**<br>
**	at is.hail.io.fs.HadoopFS.createNoCompression(HadoopFS.scala:74)**<br>
**	at is.hail.io.fs.FS$class.create(FS.scala:151)**<br>
**	at is.hail.io.fs.HadoopFS.create(HadoopFS.scala:70)**<br>
**	at is.hail.io.RichContextRDDRegionValue$.writeSplitRegion(RichContextRDDRegionValue.scala:106)**<br>
**	at is.hail.rvd.RVD$$anonfun$29.apply(RVD.scala:938)**<br>
**	at is.hail.rvd.RVD$$anonfun$29.apply(RVD.scala:936)**<br>
**	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitionsWithIndex$1$$anonfun$apply$18.apply(ContextRDD.scala:259)**<br>
**	at is.hail.sparkextras.ContextRDD$$anonfun$cmapPartitionsWithIndex$1$$anonfun$apply$18.apply(ContextRDD.scala:259)**<br>
**	at is.hail.utils.richUtils.RichContextRDD$$anonfun$cleanupRegions$1$$anonfun$2.apply(RichContextRDD.scala:62)**<br>
**	at is.hail.utils.richUtils.RichContextRDD$$anonfun$cleanupRegions$1$$anonfun$2.apply(RichContextRDD.scala:62)**<br>
**	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:435)**<br>
**	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:441)**<br>
**	at is.hail.utils.richUtils.RichContextRDD$$anonfun$cleanupRegions$1$$anon$1.hasNext(RichContextRDD.scala:71)**<br>
**	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:439)**<br>
**	at scala.collection.Iterator$class.foreach(Iterator.scala:891)**<br>
**	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)**<br>
**	at scala.collection.generic.Growable$class.$plus$plus$eq(Growable.scala:59)**<br>
**	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:104)**<br>
**	at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:48)**<br>
**	at scala.collection.TraversableOnce$class.to(TraversableOnce.scala:310)**<br>
**	at scala.collection.AbstractIterator.to(Iterator.scala:1334)**<br>
**	at scala.collection.TraversableOnce$class.toBuffer(TraversableOnce.scala:302)**<br>
**	at scala.collection.AbstractIterator.toBuffer(Iterator.scala:1334)**<br>
**	at scala.collection.TraversableOnce$class.toArray(TraversableOnce.scala:289)**<br>
**	at scala.collection.AbstractIterator.toArray(Iterator.scala:1334)**<br>
**	at org.apache.spark.rdd.RDD$$anonfun$collect$1$$anonfun$13.apply(RDD.scala:945)**<br>
**	at org.apache.spark.rdd.RDD$$anonfun$collect$1$$anonfun$13.apply(RDD.scala:945)**<br>
**	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)**<br>
**	at org.apache.spark.SparkContext$$anonfun$runJob$5.apply(SparkContext.scala:2101)**<br>
**	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)**<br>
**	at org.apache.spark.scheduler.Task.run(Task.scala:121)**<br>
**	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:403)**<br>
**	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)**<br>
**	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:409)**<br>
**	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)**<br>
**	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)**<br>
**	at java.lang.Thread.run(Thread.java:748)**</p>
<p><strong>Hail version: 0.2.64-1ef70187dc78</strong><br>
<strong>Error summary: IOException: Mkdirs failed to create /data/1kg.mt/rows/rows/parts (exists=false, cwd=file:/home/zaki)</strong></p>

<h2>danking said:</h2>
<p>You’re asking it to download the data to <code>/data</code>, meaning the root of your filesystem. In Windows terms that’s like trying to write to <code>C:\data</code>. Normal users aren’t allowed to do that. Note that the tutorial <a href="https://hail.is/docs/0.2/tutorials/01-genome-wide-association-study.html#Download-public-1000-Genomes-data">downloads the data to <code>data/</code></a>.</p>

<h2>Zaki said:</h2>
<p>So, what do you suggest i should do.</p>

<h2>danking said:</h2>
<p>Try removing the “/“ in the command you ran:</p>

```python
hl.utils.get_1kg('data')

```

<h2>Zaki said:</h2>
<p>Thank you danking its working perfectly now. <img src="https://emoji.discourse-cdn.com/twitter/+1.png?v=9" title=":+1:" class="emoji" alt=":+1:"></p>
